dxr/query.py

```diff
@@ -372,6 +413,25 @@ def filter_menu_items(plugins):
             if filter.description)
 
 
+def highlight_path(path, extents):
```

Comment:
I'm not a fan of the backtracking this does, but I can't think of an alternative that doesn't greatly increase the complexity of highlight().

So I'll just complain about the docstring, which confused me until I really studied the procedure. Something like "Return a list of path segments, with <b> tags around highlit extents." (I at first thought you mean "Return a list of path fragments that contain <b> tags"—and do something else to those that don't.)

I think this will be fine if a <b> tag opens in one segment and closes in another, but that's only because the breadcrumbs happen to be made out of all inline elements atm. It's worth a comment in the template to make sure we don't change that at some point.